packages/apps/kadena-transfer/src/services/coin/transfer-create.ts

```diff
@@ -0,0 +1,65 @@
+import { PactCommand } from '@kadena/client';
+import { sign } from '@kadena/cryptography-utils';
+
+const NETWORK_ID = 'testnet04';
+const chainId = '1';
+const gasLimit = 2300;
+const gasPrice = 0.00001;
+const ttl = 28800;
+export const API_HOST = `https://api.testnet.chainweb.com/chainweb/0.0/${NETWORK_ID}/chain/${chainId}/pact`;
+
+export async function makeTransferCreate(
+  fromAccount: string,
+  toAccount: string,
+  amount: string,
+  fromPublicKey: string,
+  toPublickKey: string,
+  fromPrivateKey: string,
+): Promise<void> {
+  const pactCommand = new PactCommand();
+  pactCommand.code = `(coin.transfer-create "${fromAccount}" "${toAccount}" (read-keyset "ks") ${amount})`;
+
+  const numberAmount = Number(amount);
+
+  pactCommand
+    .addCap('coin.GAS', fromPublicKey)
+    .addCap('coin.TRANSFER', fromPublicKey, [
+      fromAccount,
+      toAccount,
+      numberAmount,
+    ])
+    .addData({ ks: { pred: 'keys-all', keys: [toPublickKey] } })
+    .setMeta(
+      {
+        gasLimit: gasLimit,
+        gasPrice: gasPrice,
+        ttl: ttl,
+        chainId: chainId,
+        sender: fromAccount,
+      },
+      NETWORK_ID,
+    );
+
+  pactCommand.createCommand();
+
+  const signature = sign(pactCommand.cmd ?? '', {
+    secretKey: fromPrivateKey,
+    publicKey: fromPublicKey,
+  });
+
+  console.log('Signature: ', signature);
+
+  pactCommand.addSignatures({
+    pubKey: fromPublicKey,
+    sig: signature.sig ?? '',
```

Comment:
I think it's better to explicitly check it the signing went well and throw when it did not. Without signatures the transaction will fail anyhow.

Reply:
I am unable to find a button to change this from an actual PR to a draft PR. I think Nil accidentally created it as a real PR.